### PR TITLE
[kuberay][GCS GA] Add Redis dependency to ray-deps.

### LINKED ISSUE #26427: Add python redis package to ray Dockerfile
### Description

To support Ray HA. currently we need external redis. For python to connect to redis server, we need python redis package. Currently, redis package is not installed in ray image. We need to have it installed in the dockerfile

For Kubeary to support Ray HA, in our current patch, we need to install redis package https://github.com/ray-project/kuberay/pull/294

@simon-mo @iycheng @brucez-anyscale @DmitriGekhtman @Jeffwan 

### Use case

Ray HA external redis connection.

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

GCS HA requires the Redis python client to interact with an external Redis. 

## Related issue number

Closes https://github.com/ray-project/ray/issues/26427.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wilsonwang371** [comment]
lgtm.

File: docker/ray-deps/Dockerfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ COPY $FIND_LINKS_PATH $FIND_LINKS_PATH
 RUN $HOME/anaconda3/bin/pip --no-cache-dir install --find-links $FIND_LINKS_PATH \
     $(basename $WHEEL_PATH)[all] \
     $(if [ "$AUTOSCALER" = "autoscaler" ]; then echo \
+        "redis >= 3.5.0, < 4.0.0" \
```
Comment (DmitriGekhtman) [inline docker/ray-deps/Dockerfile:15]: The versions match the current [requirements.txt](https://github.com/ray-project/ray/blob/84279286df856040c772543e9d961502caaa0a70/python/requirements.txt#L88), where redis is specified for testing purposes.

**Review by brucez-anyscale — APPROVED** [review]

**Review by scv119 — APPROVED** [review]

**Review by simon-mo — APPROVED** [review]

**Comment by wilsonwang371** [comment]
is this in latest nightly build now?

**Comment by DmitriGekhtman** [comment]
Attempt #2
https://github.com/ray-project/ray/pull/26459

**Pull request merged**